metadata-db-app/src/cmr/metadata_db/migrations/074_add_normalized_to_subscription_table.clj

```diff
@@ -0,0 +1,52 @@
+(ns cmr.metadata-db.migrations.074-add-normalized-to-subscription-table
+  (:require
+   [cheshire.core :as json]
+   [cmr.common.util :as util]
+   [cmr.common-app.services.ingest.subscription-common :as sub-common]
+   [config.mdb-migrate-helper :as helper]))
+
+(defn result->query
+  "From a Database result, extract out the query and convert it to a usable string.
+   The algorithm on how to normalize the query may change over time, for this
+   migration version 1 will be used."
+  [result]
+  (-> (:metadata result)
+        util/gzip-blob->string
+        (as-> % (json/parse-string %, true))
+        :Query
+        sub-common/normalize-parameters-v1))
+
+(defn populate-new-column
+  "Pull out the requested query from the matadata, then normalize the data and
+   populate the new 'normalized_query' column."
+  []
+  (doseq [result (helper/query "SELECT id, concept_id, metadata FROM cmr_subscriptions")]
+    (let [id (:id result)
+          query (result->query result)
+          sql-statment (format "UPDATE cmr_subscriptions SET normalized_query='%s' WHERE id=%s" query id)]
+      (helper/sql sql-statment))))
+
+(defn- create-subscription-index
+  []
+  (helper/sql "CREATE INDEX cmr_subs_ccinqsi ON CMR_SUBSCRIPTIONS
+              (collection_concept_id, normalized_query, subscriber_id)"))
```

Comment:
change the order to `(collection_concept_id, subscriber_id, normalized_query)`

Reply:
I have no problems with this change, but I'm very curious why you prefer this order.

Reply:
Good question. The order in the index matters in that if you use partial fields to do a search, the index can still be used if you use the first couple fields in a multi-fields index. It is more likely for a user to search for subscriptions based on collection_concept_id and subscriber_id than collection_concept_id and normalized_query. That is why I want you to make that change. It is not something that we would definitely benefit from, but I just can't unseen it once I noticed it. :)

Reply:
I see - I will definitely keep this in mind for the future.